lib/active_model_serializers/adapter/attributes.rb

```diff
@@ -46,7 +46,8 @@ def relationship_value_for(association, options)
 
         opts = instance_options.merge(include: @include_tree[association.key])
         relationship_value = Attributes.new(association.serializer, opts).serializable_hash(options)
-        if association.options[:polymorphic]
+
+        if association.options[:polymorphic] && relationship_value
```

Comment:
@beauby good call, added check for empty relationship which wont wrap the resultant in a polymorphic type if there is no association (corresponding tests added for :json and :attributes adapters)